docs/instructor-training-workflow.txt

```diff
@@ -0,0 +1,174 @@
+  - AMY adds <Training Progress> to <Training Progress Review List>.
+
+* [trainer reports completion of written work]: on behalf of trainee.
+  - Trainer looks up <Training Progress> associated with trainee.
+  - Trainer enters URL of written work in <Training Task>.
+  - Trainer marks <Training Task> as complete.
+
+## Tracking participation in interactive work.
+
+* [trainer reports successful completion of interactive work]: such as discussion session or lesson demo.
+  - Trainer views {training progress} for trainee.
+  - Trainer selects <Training Task>.
+  - Trainer:
+    - Enters date of interactive event.
+    - Adds comments (if any).
+  - Trainer indicates whether <Training Task> is now complete.
+
+Note: each <Training Task> may accumulate multiple date+comment pairs, one for each attempt to pass that task.
+
+## Completing training work.
+
+* [trainer changes completion status of work for trainee]
+  - Trainer looks up <Training Progress> associated with trainee.
+  - Trainer changes "completed" status of work item(s).
+  => [training automatically completed]
+
+* [trainer finds work to review]
+  - Trainer views {training progress display}.
+  - Trainer selects <Training Progress>.
+  - Trainer => [trainer changes completion status of work for trainee].
+
+## Completing training.
+
+* [training automatically completed]: triggered by changes in <Training Task>s.
+  - AMY records change to <Training Task> and all <Training Task>s for <Person> now complete.
+  - AMY displays {badge creation form}.
+  - Admin confirms creation of badge.
+  - AMY moves <Training Progress> to <Past Training Progress List>.
+
+## Ending training.
+
+* [trainer ends training]: because trainee has gone silent, failed a work item, etc.
+  - Trainer views {training progress display}.
+  - Trainer selects <Training Progress>.
+  - Trainer marks <Training Progress> as "unsuccessful".
+  - AMY moves <Training Progress> to <Past Training Progress List>.
+
+* [admin ends training course]: because it has timed out.
+  - Admin selects training <Event>.
+  - Admin marks <Event> as "stalled".
+  - AMY marks all associated <Training Progress>es as "unsuccessful".
+  - AMY moves <Training Progress>es to <Past Training Progress List>.
+
+## Creating certificates.
+
+* [admin creates certificate]: for trainers who don't yet have them.
+  - Admin goes to {certificate creation form}.
+  - Admin selects <Person>s who:
+    - Have an instructor badge.
+    - Do not yet have a PDF certificate.
+  - Admin adjusts information in form.
+    - Changes name of instructor(s) awarding certificates.
+    - Changes date of award.
+  - AMY generates PDF certificates.
+  - User views/downloads certificate on {certificate display}.
+
+## Questions
+
+- How to display a mix of both <Training Application>s and <Group Training Application>s?
+- How to select from a mix of both <Training Application>s and <Group Training Application>s?
```

Comment:
It was a database question - as you pointed out, it's hard to have heterogeneous objects.